source/common/upstream/load_balancer_impl.h

```diff
@@ -465,11 +478,26 @@ class LeastRequestLoadBalancer : public EdfLoadBalancerBase {
     // be the only/best way of doing this. Essentially, it makes weight and active requests equally
     // important. Are they equally important in practice? There is no right answer here and we might
     // want to iterate on this as we gain more experience.
-    return static_cast<double>(host.weight()) / (host.stats().rq_active_.value() + 1);
+    const double weight = static_cast<double>(host.weight()) /
+                          std::pow(host.stats().rq_active_.value() + 1, active_requests_exponent_);
```

Comment:
For perf reasons I would probably check if 1.0 and then branch and avoid the pow call.